presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -68,18 +68,24 @@ public Analysis analyze(Statement statement, boolean isDescribe)
         return analysis;
     }
 
-    static void verifyNoAggregatesOrWindowFunctions(Metadata metadata, Expression predicate, String clause)
+    static void verifyNoAggregatesWindowOrGroupingFunctions(Metadata metadata, Expression predicate, String clause)
     {
         AggregateExtractor extractor = new AggregateExtractor(metadata);
         extractor.process(predicate, null);
 
         WindowFunctionExtractor windowExtractor = new WindowFunctionExtractor();
         windowExtractor.process(predicate, null);
 
-        List<FunctionCall> found = ImmutableList.copyOf(Iterables.concat(extractor.getAggregates(), windowExtractor.getWindowFunctions()));
+        GroupingOperationAnalyzer.Extractor groupingOperationExtractor = new GroupingOperationAnalyzer.Extractor();
+        groupingOperationExtractor.process(predicate, null);
+
+        List<Expression> found = ImmutableList.copyOf(Iterables.concat(
```

Comment:
`Iterables.concat` actually returns an iteration so you need some way to convert that to a list which is why I used `ImmutableList.copyOf`.